packages/monorepo-tools/src/utils/get-packages-in-topological-order.ts

```diff
@@ -17,24 +17,29 @@ export interface PackageInfo {
 }
 
 export async function getPackagesInTopologicalOrder(
-  cwd: string
+  monorepoRoot: string
 ): Promise<PackageInfo[]> {
   const patterns: string[] =
-    JSON.parse(await fs.readFile(path.join(cwd, `package.json`), 'utf8'))
-      .workspaces || [];
+    JSON.parse(
+      await fs.readFile(path.join(monorepoRoot, `package.json`), 'utf8')
+    ).workspaces || [];
 
   const packageJsonPaths = await glob(
-    patterns.map((pattern) => path.join(pattern, 'package.json'))
+    // NOTE: glob patterns should always use forward slashes,
+    // path.join here wouldn't work on win.
+    patterns.map((pattern) => `${pattern}/package.json`),
+    { cwd: monorepoRoot }
   );
 
   const internalPackages: InternalPackageInfo[] = await Promise.all(
     packageJsonPaths.map(async (packageJsonPath) => {
+      const packageJsonLocation = path.resolve(monorepoRoot, packageJsonPath);
       const packageJson = JSON.parse(
-        await fs.readFile(packageJsonPath, 'utf8')
+        await fs.readFile(packageJsonLocation, 'utf8')
       );
       return {
         name: packageJson.name,
-        location: packageJsonPath,
+        location: path.dirname(packageJsonLocation),
```

Comment:
This was previously a filename and is now a directory name, just making sure that that is intentional?

Reply:
Kinda. We only ever used it as a dirname and by the time we return to the outside we just want a path to a directory. Let me clean that up so it is clearer.